### DISCUSSION TITLE
Client Write UDT

### DISCUSSION THREAD

**Discussion by alattanzi** [eclipse-milo/milo#1132] [General]
Hi All,

In my plc I have created a UDT with 2 boolean field.
I have registered the coded successful. I'm able to read and decode the structure, but I have some issue when I try to write it.
In the details, I receive this error
StatusCode{name=Bad_TypeMismatch, value=0x80740000, quality=bad}
I try to write it in this way
![image](https://github.com/eclipse/milo/assets/137071485/5f3dc6f0-48d7-46ae-ab97-2fa19d979434)

and also in this 
![image](https://github.com/eclipse/milo/assets/137071485/83a60b43-aa41-4310-977c-505c84a4b4d3)

but the result it is always the same.
This is my codec class
![image](https://github.com/eclipse/milo/assets/137071485/8375b986-cf52-4775-954d-f87320ab5e15)

where the decode is working.

Thank you
Andrea

**Comment by kevinherron** [comment]
Do you have anything that shows the definition in the PLC?

Can you get a Wireshark capture? Does writing from another client like UaExpert work?

**Comment by alattanzi** [comment]
This is my UDT in the PLC
![image](https://github.com/eclipse/milo/assets/137071485/de345409-0381-4dce-b200-04d74fd13c1f)

and from UA expert I can modify the value
![image](https://github.com/eclipse/milo/assets/137071485/248f580c-2d84-4316-81f1-e3cf7b3b3ab6)

and this is the Wireshark captured for the write request
![image](https://github.com/eclipse/milo/assets/137071485/c68666a7-81a5-40ce-99a5-830989ae580f)

Thanks

**Reply by kevinherron** [reply]
Sorry, can you upload both a Wireshark capture of the write from UaExpert and the write from Milo so I can look at the details and compare them?

**Comment by alattanzi** [comment]
Here the 2 file.
[WS_Milo.txt](https://github.com/eclipse/milo/files/11814624/WS_Milo.txt)
[WS_UaExpert.txt](https://github.com/eclipse/milo/files/11814625/WS_UaExpert.txt)

Thank you
Andrea

**Comment by kevinherron** [comment]
The capture shows that you are using a different NodeId for the TypeId than UaExpert is:

Milo:
```
                                    Namespace Index: 3
                                    Identifier String: TD_"UDT1000"
```

UaExpert:
```
                                    Namespace Index: 4
                                    Identifier Numeric: 239
```

**Comment by alattanzi** [comment]
Thank you for your help.
Now it is working.

The Node that you see in Milo is the one with the UDT structure in the PLC
The other is the encoder used for the server.

Now i go to set the nodeId of the encoder and I can write the signal

Thanks again
Andrea
